cloudwatchlogs/node.js/cloudwatchlogs_lambda.js

```diff
@@ -2,79 +2,112 @@
+                    body += chunk;
+                });
+                res.on('end', function () {
+                    console.log('Successfully processed HTTPS response');
+                    requestsSent++;
+                    finalizeContext();
+                });
+            });
+            
+            req.on('error', function (e) {
+                console.log(e.message);
+                requestsFailed++;
+                finalizeContext();
+            });
+            
+			// Remove trailing \n
+			val.message = val.message.replace(/\n$/, '');
+			
+            // Try extract requestID
+            var requestId = requestIdRegex.exec(val.message);
+            if (requestId !== null) {
+                lastRequestID = requestId[1];
+            }
+            
+            // Attempt to detect console log and auto extract requestID and message
+            var consoleLog = consoleFormatRegex.exec(val.message);
+            if (consoleLog !== null) {
+                lastRequestID = consoleLog[1];
+                val.message = consoleLog[2].trim();
+            }
 
-			var awslogsData = JSON.parse(buffer.toString('ascii'));
+            // Auto detect if message is json
+            try {
+                val.message = JSON.parse(val.message);
+            } catch (err) {
+                // Do nothing, leave as text
+                val.message.trim()
+            }
+			
+            // delete id as it's not very useful
+            delete val.id;
 
-			console.log(awslogsData);
-
-			if (awslogsData.messageType === "CONTROL_MESSAGE") {
-				console.log("Control message");
-				context.succeed("Success");
-			}
-
-			var requestsSent = 0;
-			var requestsFailed = 0;
-			var finalizeContext = function() {
-				var tot = requestsSent + requestsFailed;
-				if (tot == awslogsData.logEvents.length) {
-					if (requestsFailed > 0) {
-						context.fail(requestsFailed + " / " + tot + " events failed");
-					} else {
-						context.succeed(requestsSent + " requests sent");
-					}
-				}
-			};
-
-			var re = new RegExp(/(?:RequestId:|Z)\s+([\w\d\-]+)/);
-			var lastRequestID = null;
-			awslogsData.logEvents.forEach(function(val, idx, arr) {
-				var req = https.request(options, function(res) {
-					var body = '';
-					console.log('Status:', res.statusCode);
-					res.setEncoding('utf8');
-					res.on('data', function(chunk) { body += chunk; });
-					res.on('end', function() {
-						console.log('Successfully processed HTTPS response');
-						requestsSent++;
-						finalizeContext();
-					});
-				});
-
-				req.on('error', function(e) {
-					console.log(e.message)
-					requestsFailed++;
-					finalizeContext();
-				});
-
-				var stream=awslogsData.logStream;
-				var group=awslogsData.logGroup;
-				var rs = re.exec(val.message);
-				if (rs!==null) { 
-                		   lastRequestID = rs[1]; 
-                		}
-				val.requestID = lastRequestID;
-				val.logStream = stream;
```

Comment:
@stevezau : we need to make logStream and logGroup on by default, otherwise data here will not be compatible with our Lambda app.